capgains/commands/capgains_calc.py

```diff
@@ -60,15 +62,39 @@ def _filter_calculated_transaction(transaction, year):
     return True
 
 
+def _get_map_of_currencies_to_exchange_rates(transactions):
+    currency_date_ranges = dict()
+    # First, map the currency to a range of dates it needs
+    for transaction in transactions:
+        currency = transaction.currency
+        date = transaction.date
+        if currency not in currency_date_ranges:
+            currency_date_ranges[currency] = (date, date)
+        elif date < currency_date_ranges[currency][0]:
+            currency_date_ranges[currency] = \

```

Comment:
Add a comment saying that this is setting the minimum date.

Reply:
Fixed